### DISCUSSION TITLE
wrong caret position on moving left with inlined decoration

### DISCUSSION THREAD

**Discussion by agusakow** [microsoft/monaco-editor#3772] [Q&A] [answered]
Does this issue occur when all extensions are disabled?: Yes

- Monaco Version: 0.31.0, Code OSS Dev: 1.63.0

- OS Version: 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/97.0.4692.20 Safari/537.36'

Steps to Reproduce:

1. Open https://microsoft.github.io/monaco-editor/playground.html
2. Paste code
```javascript
var editor = monaco.editor.create(document.getElementById('container'), {
	value: "function hello() {\n\talert('Hello world!');\n}",
	language: 'javascript'
});

editor.getModel().deltaDecorations([], [{
        range: {
		startLineNumber: 1,
		endLineNumber: 1,
		startColumn: 1 << 30,
		endColumn: 1 << 30
	},
	options: {
		after: {
			content: "Inlined text",
			inlineClassNameAffectsLetterSpacing: true,
		},
		showIfCollapsed: true
	}
}]);
```
3. Paste cursor at the beginning of 2 line
4. Press left

Actual cursor position is :

```javascript
function hello() {Inlined text|
	alert('Hello world!');
}
```

Expected:
```javascript
function hello() {|Inlined text
	alert('Hello world!');
}
```

**Comment by alexdima** [comment, marked as answer]
This is the current behavior of inserted text, it creates two distinct view cursor stops (one before the inserted text and one after the inserted text). You can discover the two distinct view cursor stops by clicking with the mouse. Both view cursor stops map to the same model cursor position, therefore we have decided to skip one when moving left/right. When moving left, we stop at the second one and skip the first one. When moving right, we stop at the first one and skip the second one. IMHO we could add more API such that this can be made configurable by the creator of the inserted text.

cc @hediet

**Comment by agusakow** [comment]
Thank you for your explanations!
I'm trying to implement something like inline variable values at VsCode debugger with this injected text API.

This is behavior from VsCode Version: 1.62.3:
![](https://i.imgur.com/J4yqsTV.gif)


And this is from Code OSS Dev: 1.63.0:
![](https://i.imgur.com/CtqlBi6.gif)


Am I right that the latest version use injected text API and this is the intended behavior?

**Reply by KurtGokhan** [reply]
Did you find a solution to this? It seems wrong to me and should be fixed imho.

**Comment by y912765390** [comment]
Annotations are not allowed to register JSON directly If I want to use comments in JSON without reporting errors, what should I do.

If  we can use "setcompileroptions" So which configuration is it
